parseAndPopulate/modulesComplicatedAlgorithms.py

```diff
@@ -630,97 +635,83 @@ def __parse_semver(self):
                                 self.__new_modules[name_revision]['derived-semantic-version'] = module['derived-semantic-version']
                             continue
                         else:
```

Comment:
**NOTE**: Following code was not deleted, only the if condition was added and code was indented